### PR TITLE
[wiring] acquireSerialXBuffer() support

### PR DESCRIPTION
**Pull request by avtolstoy**
### Problem

The default 64-byte USART RX/TX buffer may not be sufficient for some applications.

### Solution

1. Introduce API similar to https://docs.particle.io/cards/firmware/serial/acquireserialbuffer/ or https://docs.particle.io/cards/firmware/wire-i2c/acquirewirebuffer/ for USARTs
2. Introduce `hal_usart_init_ex()` support for Gen 2

### Steps to Test

- `wiring/serial_loopback`
- `wiring/serial_loopback2`

Build tests with default settings and with `USE_BUFFER_SIZE=256` for example.

### Example App

N/A

### References

N/A

---

### Completeness

- [x] User is totes amazing for contributing!
- [ ] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [ ] Problem and Solution clearly stated
- [ ] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

File: user/tests/wiring/serial_loopback/serial.cpp [hunk 1]
```diff
@@ -26,6 +26,29 @@
 #include "application.h"
 #include "unit-test/unit-test.h"
 
+#ifndef USE_BUFFER_SIZE
+#warning "Using default 64 byte buffer"
+#define USE_BUFFER_SIZE SERIAL_BUFFER_SIZE
+#else
+hal_usart_buffer_config_t __attribute__((weak)) acquireSerial1Buffer()
```
Comment (XuGuohui) [inline user/tests/wiring/serial_loopback/serial.cpp:33]: `__attribute__((weak))` needs to be removed?

File: user/tests/wiring/serial_loopback2/loopback.cpp [hunk 2]
```diff
@@ -19,15 +19,45 @@
 #include "unit-test/unit-test.h"
 #include "random.h"
 
-void runLoopback(size_t buffer_size_min, size_t buffer_size_max, bool sleep) {
+#ifndef USE_BUFFER_SIZE
+#warning "Using default 64 byte buffer"
+#define USE_BUFFER_SIZE (SERIAL_BUFFER_SIZE)
+#else
+hal_usart_buffer_config_t __attribute__((weak)) acquireSerial1Buffer()
```
Comment (XuGuohui) [inline user/tests/wiring/serial_loopback2/loopback.cpp:26]: `__attribute__((weak))` needs to be removed?

File: user/tests/wiring/serial_loopback/serial.cpp (hunk 1 above)
Comment (avtolstoy) [reply user/tests/wiring/serial_loopback/serial.cpp]: :+1: fixed

File: user/tests/wiring/serial_loopback2/loopback.cpp (hunk 2 above)
Comment (avtolstoy) [reply user/tests/wiring/serial_loopback2/loopback.cpp]: :+1: fixed

**Review by XuGuohui — APPROVED** [review]

**Review by keeramis — APPROVED** [review]
Ran on-device tests on Electron and Boron, and they passed.

**Pull request merged**